src/include/txn_inline.h

```diff
@@ -1789,17 +1785,17 @@ __wt_txn_activity_check(WT_SESSION_IMPL *session, bool *txn_active)
 static inline void
 __wt_upd_value_assign(WT_UPDATE_VALUE *upd_value, WT_UPDATE *upd)
 {
-    if (!upd_value->skip_buf) {
-        upd_value->buf.data = upd->data;
-        upd_value->buf.size = upd->size;
-    }
     if (upd->type == WT_UPDATE_TOMBSTONE) {
         upd_value->tw.durable_stop_ts = upd->durable_ts;
         upd_value->tw.stop_ts = upd->start_ts;
         upd_value->tw.stop_txn = upd->txnid;
         upd_value->tw.prepare =
           upd->prepare_state == WT_PREPARE_INPROGRESS || upd->prepare_state == WT_PREPARE_LOCKED;
     } else {
+        if (!upd_value->skip_buf) {
```

Comment:
My reasoning for moving this block is that we'd only want to assign these values when there is a `data` and `size` which there won't be with a tombstone.